packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -111,25 +113,58 @@ const ProductDetail = ({category, product, isLoading}) => {
             status: 'error'
         })
     }
-    const handleAddToCart = async (variant, quantity) => {
+
+    const handleAddToCart = async (productSelectionValues) => {
         try {
-            if (!variant?.orderable || !quantity) return
-            // The basket accepts an array of `ProductItems`, so lets create a single
-            // item array to add to the basket.
-            const productItems = [
-                {
-                    productId: variant.productId,
-                    quantity,
-                    price: variant.price
-                }
-            ]
+            const productItems = productSelectionValues.map(({variant, quantity}) => ({
+                productId: variant.productId,
+                price: variant.price,
+                quantity
+            }))
 
             await basket.addItemToBasket(productItems)
+
+            // If the items were sucessfully added, set the return value to be used
+            // by the add to cart modal.
+            return productSelectionValues
         } catch (error) {
             showError(error)
         }
     }
 
+    /**************** Product Set Handlers ****************/
+    const handleProductSetValidation = () => {
+        // Scroll to the first product without its options selected.
+        const productSelectionKeys = Object.keys(productSetSelection)
+        const unselected = product.setProducts.find(({id}) => !productSelectionKeys.includes(id))
+
+        // Run "internal" validation on this product to show the error.
+        Object.values(childProductRefs.current).forEach(({scope}) => {
+            scope.validateAttributeSelection()
+        })
```

Comment:
I think you wanted to have this block inside `if (unselected)` below, right?

Reply:
Actually, that is how I intended it to be called. To validate the product set, I validate all the children. It just so happens that I'm also tracking which children have ultimately been selected. So I could technically only validate once I know there is some children that aren't selected, or even go even further and add logic to only validate those that aren't selected. 

But this goes against what the original intention for this function which is to act like a form validating it's inputs.

Alternatively I could have the child validation return some type of error object, and I think that ultimately the product view should have been developed more like a form, but that isn't how it's done, so it would be a lot of effort to do that now. 

I've gone ahead and made the code a little more easy to read so it's more clear what is happening here.

Reply:
Lets talk monday to see if there is a better way we can handle all this validation and scrolling stuff.